dietpi/dietpi-software

```diff
@@ -9055,33 +9056,34 @@ _EOF_
 
 			Banner_Configuration
 
-			if [[ ! -f '/opt/AdGuardHome/AdGuardHome.yaml' ]]
+			# Directory
+			G_EXEC mkdir -p /mnt/dietpi_userdata/adguardhome
+
+			# User
+			Create_User -d /mnt/dietpi_userdata/adguardhome adguardhome
+
+			if [[ ! -f '/mnt/dietpi_userdata/adguardhome/AdGuardHome.yaml' ]]
 			then
-				dps_index=$software_id Download_Install 'AdGuardHome.yaml' /opt/AdGuardHome/AdGuardHome.yaml
+				dps_index=$software_id Download_Install 'AdGuardHome.yaml' /mnt/dietpi_userdata/adguardhome/AdGuardHome.yaml
 				# shellcheck disable=SC2016
-				G_CONFIG_INJECT 'password:[[:blank:]]' "  password: $(htpasswd -bnBC 10 '' "$GLOBAL_PW" | tr -d ':\n' | sed 's/\$2y/\$2a/')" /opt/AdGuardHome/AdGuardHome.yaml
+				G_CONFIG_INJECT 'password:[[:blank:]]' "  password: $(htpasswd -bnBC 10 '' "$GLOBAL_PW" | tr -d ':\n' | sed 's/\$2y/\$2a/')" /mnt/dietpi_userdata/adguardhome/AdGuardHome.yaml
 			fi
 
 			# Unbound: Configure AdGuard Home to use it
 			if (( ${aSOFTWARE_INSTALL_STATE[182]} > 0 ))
 			then
 				G_DIETPI-NOTIFY 2 'Configuring AdGuard Home to use Unbound'
-				if [[ ! -f '/opt/AdGuardHome/dietpi-unbound.conf' ]]
+				if [[ ! -f '/mnt/dietpi_userdata/adguardhome/dietpi-unbound.conf' ]]
 				then
-					echo '127.0.0.1:5335' > /opt/AdGuardHome/dietpi-unbound.conf
-					G_CONFIG_INJECT 'upstream_dns_file:[[:blank:]]' '  upstream_dns_file: /opt/AdGuardHome/dietpi-unbound.conf' /opt/AdGuardHome/AdGuardHome.yaml
+					echo '127.0.0.1:5335' > /mnt/dietpi_userdata/adguardhome/dietpi-unbound.conf
+					G_CONFIG_INJECT 'upstream_dns_file:[[:blank:]]' '  upstream_dns_file: /mnt/dietpi_userdata/adguardhome/dietpi-unbound.conf' /mnt/dietpi_userdata/adguardhome/AdGuardHome.yaml
 					systemctl -q is-active adguardhome && G_EXEC systemctl restart adguardhome
 				fi
 			fi
 
-			# Directory
-			G_EXEC mkdir -p /mnt/dietpi_userdata/adguardhome
-
-			# User
-			Create_User -d /mnt/dietpi_userdata/adguardhome adguardhome
-			
 			# Permissions
-			chown -R adguardhome:adguardhome /mnt/dietpi_userdata/adguardhome /opt/AdGuardHome/
+			chown -R adguardhome:adguardhome /mnt/dietpi_userdata/adguardhome /opt/adguardhome
+			chmod 755 /opt/adguardhome{,/AdGuardHome}
```

Comment:
We could use `G_EXEC` here and in a few more cases.

Reply:
😃

Reply:
probably to much now 🙈

Reply:
Actually there is no reason to not use it everywhere, where no redirect is involved (which is difficult to pass to a function). That way it's also more transparent what we're actually doing during the installs. All fine for me. On uninstalls I used to use `G_EXEC_NOEXIT=1` currently to allow ignoring errors, but in the cases here, there would be a critical issues present if any of those fails, so IMO fine to stay strict.